### ISSUE TITLE
Need stock firmware 

### ISSUE THREAD

**Issue by admiralroflknife** [nekorevend/esphome-emporia-vue-utility#2]
I flashed the new software prior to having it connected to the meter by my power company. I'm not sure if I need to first use the emporia software prior and get it paired before flashing this or not, but it seems like it may be necessary. Unfortunely, my backup bin did  not save correctly so I'm stuck.

Do I need to flash back or will it start working on its own eventually? if i need the stock, can anyone share a v2 firmware file for me to download?

**Comment by nekorevend** [comment]
TL;DR, your power company should still be able to provision it with your meter even when running the custom firmware.

From what we know about the protocol between the zigbee/MGM chip and the ESP32 is the ESP32 sends a "Join the meter" message to the MGM chip with no meaningful payload. This custom firmware already sends that message periodically if meter readings fail, so it should be just as good at getting provisioned as the stock firmware would.

Otherwise, I've never had a Vue with the V2 firmware, sorry. You could try asking in the Home Assistant thread: https://community.home-assistant.io/t/emporia-vue-utility-connect/378347

P.S. For backing up original firmware (in general, for every device), I always do it twice and check if the two backup files are identical (via checksum). It's very unlikely for corruption to happen twice in the exact same spot so if any two backups match then that should be the good copy.

**Comment by admiralroflknife** [comment]
Oh that’s good to hear. I’m waiting still as I don’t think it’s an
automated process and the holiday weekends slowing it down. Hopefully on
Tuesday it’ll connect

**Comment by admiralroflknife** [comment]
yup, connected perfectly anyways. looks like i jumped the gun
